sound/soc/sof/loader.c

```diff
@@ -192,6 +192,55 @@ static int ext_man_get_dbg_abi_info(struct snd_sof_dev *sdev,
 	return 0;
 }
 
+static int ext_man_get_config_data(struct snd_sof_dev *sdev,
+				   const struct sof_ext_man_elem_header *hdr)
+{
+	const struct sof_ext_man_config_data *config =
+		container_of(hdr, struct sof_ext_man_config_data, hdr);
+
+	/* save pointer to adsp config data to make it accessible from any place in code */
+	sdev->adsp_config = config;
+	return 0;
+}
+
+static int get_config_value(const struct sof_ext_man_config_data *config, uint32_t token,
+			    uint32_t *value)
+{
+	const struct sof_config_elem *elem;
+	const struct sof_config_elem *last;
+	int elems_counter;
+	int elems_size;
+
+	/* client should log error message or set default value */
+	if (!config || config->hdr.size < sizeof(struct sof_ext_man_elem_header))
+		return -EINVAL;
+
+	/* calculate last element pointer */
+	elem = config->elems;
+	elems_size = config->hdr.size - sizeof(struct sof_ext_man_elem_header);
+	last = &elem[elems_size / sizeof(struct sof_config_elem)];
+
+	while (elem != last) {
```

Comment:
since you'll have to resubmit the series because of the unused variable above, you might as well consider making this a `for ()` loop

Reply:
changed to `for`